runtime/oti/VMHelpers.hpp

```diff
@@ -2048,18 +2049,56 @@ class VM_VMHelpers
 #endif /* JAVA_SPEC_VERSION > 11 */
 	}
 
-	static VMINLINE UDATA
-	walkContinuationStackFramesWrapper(J9VMThread *vmThread, j9object_t continuationObject, J9StackWalkState *walkState)
-	{
-		UDATA rc = J9_STACKWALK_RC_NONE;
 #if JAVA_SPEC_VERSION >= 19
-		J9VMContinuation *continuation = J9VMJDKINTERNALVMCONTINUATION_VMREF(vmThread, continuationObject);
-		rc = vmThread->javaVM->internalVMFunctions->walkContinuationStackFrames(vmThread, continuation, walkState);
-#endif /* JAVA_SPEC_VERSION >= 19 */
-		return rc;
+/*
+ *	If low tagging failed (or not even attempted) due to either
+ *
+ *   a carrier thread winning to mount, we don't need to do anything, since it will be compensated by pre/post mount actions
+ *   another GC thread winning to scan, again don't do anything, and let the winning thread do the work, instead
+ */
+	static VMINLINE bool
+	tryWinningConcurrentGCScan(J9VMContinuation *continuation)
+	{
+		volatile uintptr_t *localAddr = &continuation->state;
+		uintptr_t oldContinuationState = *localAddr;
+
+		if (0 == oldContinuationState) {
+			/* atomically low tag ('set', not 'or') continuation->state */
+			if (VM_AtomicSupport::lockCompareExchange(localAddr, oldContinuationState, J9_GC_CONTINUATION_CONCURRENTSCANNING) == oldContinuationState) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	static VMINLINE J9VMThread *
+	getCarrierThreadFromContinuationState(uintptr_t continuationState)
+	{
+		return (J9VMThread *)(continuationState & (~(uintptr_t)J9_GC_CONTINUATION_CONCURRENTSCANNING));
+	}
+
+	static VMINLINE bool
+	isConcurrentlyScanned(uintptr_t continuationState)
```

Comment:
on a fence whether this should be called is(get?)ConcurrentlyScannedFromContinuationState